### PR TITLE
tooling: Update pipcheck tool

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: tooling: Update pipcheck tool
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
coverage is here https://storage.googleapis.com/envoy-pr/16033/tooling/index.html

File: tools/dependency/pip_check.py [hunk 1]
```diff
@@ -51,44 +43,57 @@ def config_requirements(self) -> set:
...
+    def requirements_filename(self) -> str:
+        return self._requirements_filename
 
-    def check_dependabot(self) -> int:
+    def check_dependabot(self) -> None:
         """Check that dependabot config matches requirements.txt files found in repo"""
         missing_dirs = self.config_requirements - self.requirements_dirs
         missing_config = self.requirements_dirs - self.config_requirements
-
+        correct = self.requirements_dirs & self.config_requirements
```
Comment (htuch) [inline tools/dependency/pip_check.py:69]: `&` or `and`?

Comment (phlax) [reply tools/dependency/pip_check.py:69]: these are set operations - this a set intersection, and is correct 

i could use the set methods instead i guess

File: tools/dependency/pip_check.py [hunk 2]
```diff
@@ -1,44 +1,36 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
+
+# usage
+#
+# with bazel:
+#
+#  bazel //tools/dependency:pip_check -- -h
+#
+# alternatively, if you have the necessary python deps available
```
Comment (phlax) [inline tools/dependency/pip_check.py:9]: no deps

Comment (phlax) [reply tools/dependency/pip_check.py:9]: hmm - actually there will be deps when we add pip_compile so rather leave

File: tools/dependency/pip_check.py (hunk 1 above)
Comment (htuch) [reply tools/dependency/pip_check.py:69]: I think I would probably find it a bit clearer TBH.

**Review by htuch — COMMENTED** [review]
LGTM modulo the small nit on set syntax.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

File: tools/dependency/tests/test_pip_check.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
+from unittest.mock import patch, PropertyMock
+
+import pytest
+
+from tools.dependency import pip_check, pytest_pip_check
+
+
+def test_pytest_pip_check():
```
Comment (phlax) [inline tools/dependency/tests/test_pip_check.py:8]: remove this

**Comment by phlax** [comment]
/wait

this needs rebase with #16061

**Review by htuch — APPROVED** [review]

**Pull request merged**